openvm/src/bus_interaction_handler/mod.rs

```diff
@@ -91,6 +112,131 @@ impl<T: FieldElement> IsBusStateful<T> for OpenVmBusInteractionHandler<T> {
     }
 }
 
+impl<T: FieldElement> RangeConstraintHandler<T> for OpenVmBusInteractionHandler<T> {
+    fn pure_range_constraints<V: Ord + Clone + Eq>(
+        &self,
+        bus_interaction: &BusInteraction<GroupedExpression<T, V>>,
+    ) -> Option<RangeConstraintMap<T, V>> {
+        let bus_id = bus_interaction
+            .bus_id
+            .try_to_number()
+            .unwrap()
+            .to_integer()
+            .try_into_u64()
+            .unwrap();
+        match self.bus_map.bus_type(bus_id) {
+            BusType::ExecutionBridge | BusType::Memory | BusType::PcLookup => None,
+            BusType::OpenVmBitwiseLookup => {
+                bitwise_lookup_pure_range_constraints(&bus_interaction.payload)
+            }
+            BusType::Other(OpenVmBusType::VariableRangeChecker) => {
+                variable_range_checker_pure_range_constraints(&bus_interaction.payload)
+            }
+            BusType::Other(OpenVmBusType::TupleRangeChecker) => {
+                tuple_range_checker_pure_range_constraints(&bus_interaction.payload)
+            }
+        }
+    }
+
+    fn batch_make_range_constraints<V: Ord + Clone + Eq + Display + Hash>(
+        &self,
+        mut range_constraints: BTreeMap<GroupedExpression<T, V>, RangeConstraint<T>>,
+    ) -> Vec<BusInteraction<GroupedExpression<T, V>>> {
+        let mut byte_constraints = filter_byte_constraints(&mut range_constraints);
+        let tuple_range_checker_ranges = tuple_range_checker_ranges::<T>();
+        assert_eq!(
+            tuple_range_checker_ranges.0,
+            RangeConstraint::from_mask(0xffu64),
+        );
+
+        // The tuple range checker bus can range-checks two expressions at the same time.
+        // We assume the first range is a byte range (see assertion above). From the remaining
```

Comment:
I'm a bit lost about this "first range is a byte range". First range of what? That function doesn't take any arguments